astroquery/esa/xmm_newton/core.py

```diff
@@ -94,48 +102,44 @@ def download_data(self, observation_id, *, filename=None, verbose=False,
             file format, optional, by default all formats
             values: ASC, ASZ, FTZ, HTM, IND, PDF, PNG
 
-
         Returns
         -------
         None if not verbose. It downloads the observation indicated
         If verbose returns the filename
         """
-        if filename is not None:
-            filename = os.path.splitext(filename)[0]
 
-        link = self.data_aio_url + "obsno=" + observation_id
+        # create url to access the aio
+        link = self._create_link(observation_id, **kwargs)
 
-        link = link + "".join("&{0}={1}".format(key, val)
-                              for key, val in kwargs.items())
+        # If the user wants to access proprietary data, ask them for there credentials
+        if prop:
+            username, password = self._get_username_and_password(credentials_file)
+            link = f"{link}&AIOUSER={username}&AIOPWD={password}"
```

Comment:
Note that this can result in plaintext logging/output of the username and password.  Does the server allow these data to be sent via post request rather than through the URL?

Reply:
See https://github.com/astropy/astroquery/pull/2251/checks?check_run_id=4559276557 - it's the auto-security-checker that is noting that there is plaintext logging of passwords happening

Reply:
The server that the xmm_newton archive is currently using does not handle post request. We understand that sending a plaintext username and password over url is not safe. This is the reason we have updated the http requests to be https. Once the functionality of downloading proprietary data has been implemented. The xmm team have planned to improve the aio server and change the way the user can login and access data. However until them we only have url commands.